### ISSUE TITLE
Options not working

### ISSUE THREAD

**Issue by Prb4000** [cubecart/v6#3433]
I am not using any plugins that should interfere with this, standard CubeCart and using unchanged foundation skin.
Website Butlersprintedmodels.co.uk

This is an old problem that I reported to Havenswift some time ago.  Ian has suggested I report it here:

When a product has several sets of options there is a fault with the way the website works.
I usually prefer not to set default options because I would like the customer to be prompted to choose options.

The prompting works fine and the website won't add a product to the basket when the product is added from the product page.

The problem occurs if the customer adds the product from the list of products in a category. When a customer does this there are a number of possible things that the website does.
1. If there is on only one compulsary option then the website works fine, it refuses to add the product, takes the customer to the product page and asks for the option to be chosen.

    If there is more than one compulsary option then the website chooses an option from each compulsary option set.

    However the most common thing I see is for the website to choose an option from only one of the compulsary option sets, thus adding a product to the basket with options not chosen.

Order 220318-093611-3544 has clearly been created by adding to basket from the category list page. Product 06Bri054 has had both compulsary options picked by the website.
Product 06Bri052 is missing an option
Product 06Bri058 has defaults options set and so it has worked.

Try product 15Bri052 - it works because it only has one compulsary option.

One work around I have is to set default selections for all products with two compulsary option sets. This isn't what I want to do because some customers won't be prompted to choose options. However if I don't do this we are going to keep receiving orders with products that don't have all the options chosen.

We are slowly adding a second compulsary option to all products and this is why it has now become a bigger issue for us.
The alternative work around is for me to add this new option as none compulsary. It will avoid the problem but I chose not to add it this way because I thought it was more confusing for the customer and I really wanted them to be forced to make a decision.
(try product 15Bri054, it has a compulsary and a non compulsary option and works fine).

and after using the work around for some time

I have started using a work around to the previous problem, I have changed one of the options from compulsary to non-compulsary but made one of the choices a default. It isn't ideal but it is functional and stops the previous error.

However this has now resulted in a new error with options becoming apparent. I actually think it is probably the same error as before manifesting itself in a different way.

If the product in question only has one non-compulsary option with one of the choices set as default.
If the product is added to the basket from the product page then everything works as expected 

The problem:
If the product is added from the category list view the default non-compulsary option is missing in the basket. 

This can even allow two products in the basket that are identical products with identical options but because of the error they are showing as different.

I can live with this fault because at least I know what the missing option is because it is a default option. Unfortunetly it doesn't show up for the customer.

The product code for an example is: 06Air016
Search for "Air016" for a list view and add any of the products to basket they all show the same problem.

**Comment by abrookbanks** [comment]
From my testing CubeCart will only add an item to the basket from the category list if;

- All options are set to required
- All assigned options to the product have a default value set

Arguably this is good behaviour but maybe the customer need to be informed that this has happened and given the chance to see available options. 

I think in your case the solution is to mare sure one or more assigned options is *not* set to default.

**Comment by abrookbanks** [comment]
@Prb4000 do my findings above match your?

**Comment by Prb4000** [comment]
Perhaps this will help explain part of the problem
A picture is attached.

I have added the same product to the basket.  There is a "Quality" option and it has a default value of "Standard (PLA)".
One product shows the option and one does not.

The one that does not show the option was added from a list of products rather than the actual product page.
![Options not working](https://github.com/cubecart/v6/assets/67736774/907eb9f3-18e1-41d9-916d-d1e8c1086c6e)
